### DISCUSSION TITLE
Force main domain

### DISCUSSION THREAD

**Discussion by michael** [vercel/next.js#28164] [Help] [answered]
Hi there!

I just launched a writing platform and was able to do so in about 6 months, thanks to Next.js! It's been a fantastic development experience!

I'm currently hosting on Digitalocean's App Platform and **letsken.com**, **www.letsken.com** and **beta.letsken.com** all point to the app. While I want all urls to work, I'd like to force the root domain as being the "true" location.

In other words, `https://www.letsken.com/path/to` should forward to `https://letsken.com/path/to` no matter if it the page is SSG or or SSR. Is there a way to accomplish this within Next.js where all domains serve the same app? Like without setting up separate redirect servers for beta.letsken.com and www.letsken.com....

I understand that I can setup redirects in next.config.js, but I can't find a way to set the destination to a different domain. So I guess that won't help me. Maybe it could be solved with an Express middleware that's put in place for all pages in a central place, but then I'm not sure if that middleware is even called for static pages? I also want this to work in a true serverless environment in the future. I'm planning to move from DigitalOcean to Vercel sometime soon.

Thank you very much for your help!

**Comment by michael** [comment]
Does anyone have any pointers on this? I see that @vercel (www.vercel.com/path/to) is doing exactly what I want. Cc @leerob @rauchg

Help greatly appreciated! Thank you :)

**Comment by leerob** [comment, marked as answer]
> I understand that I can set up redirects in next.config.js, but I can't find a way to set the destination to a different domain.

You can redirect to external domains! But in this case, this configuration would need to happen inside of DigitalOcean.

> I'm planning to move from DigitalOcean to Vercel sometime soon.

More information here how Vercel handles this: https://vercel.com/docs/custom-domains#redirecting-www-domains

**Reply by michael** [reply]
Thank you Lee! Unfortunately DO doesn't have such a handy configuration option on App Platform. Will contact their support again, but do not have too much hope.

Main reason I can't switch to vercel yet, is that I haven't found an easy way to have database and serverless functions in the same region. I'm currently using Frankfurt as the region for the DB Cluster, but there's no AWS region for Frankfurt available at Vercel right?

Are there any plans to support Databases (such as Postgres) in Vercel directly? That would make things soo much easier. :) Especially connection pooling and that stuff, is quite a hassle. Couldn't find any alternatives to DigitalOcean for a managed PostgresDB that can work in a serverless env.

**Reply by leerob** [reply]
Frankfurt isn't available (yet), but you could pick the next nearest and still likely have low latency 😄 

No plans to build our own managed Postgres, but we do have first-class integrations with many databases, allowing you to deploy your app in a few minutes! For example, with PlanetScale/Supabase/Fauna, you can consider them "connectionless" - connecting pooling isn't an issue, as database queries are made over HTTP! They provide a database as an API.

https://vercel.com/integrations#databases

**Reply by michael** [reply]
> Frankfurt isn't available (yet), but you could pick the next nearest and still likely have low latency 😄

When I get some time, I will try putting both to London. It's a different data center (lhr-1 and lon-1), but that's as close as I can get them atm. :)

> No plans to build our own managed Postgres, but we do have first-class integrations with many databases, allowing you to deploy your app in a few minutes! For example, with PlanetScale/Supabase/Fauna, you can consider them "connectionless" - connecting pooling isn't an issue, as database queries are made over HTTP! They provide a database as an API.
> 
> https://vercel.com/integrations#databases

Interesting space. I'm watching closely. I had Supabase and Prisma on the radar but wanted to stay close to the metal with SQL and Postgres, which I am familiar with. Also played with Amazon Aurora Serverless to get rid of "connections". Wasn't too happy with the DX though, so I concluded with Postgres on DigitalOcean and Connection pooling. See: https://github.com/vercel/vercel/discussions/5696

Out of curiosity, are you (Vercel) also hiring part-time remote positions?

**Reply by michael** [reply]
Oh, and one idea I had for a short-term fix regarding the domain forward: Would it work, if I plugged in an express middleware (I'm using a regular Node.js server instance atm)?

Till i switch to Vercel...
